ONSdigital/eq-author-app
# EAR 2094 list answer error

> ### BEFORE MAKING YOUR PR
>
> Please ensure:
>
> - There are no linting errors, all tests must pass
> - PR is named after JIRA ticket number e.g. EAR-###
> - **Accesibility** checks are completed:
>   - Elements have discernible and consistent focus states
>   - Elements can be navigated to and used by just a keyboard
>   - Elements and text can be read out by a screen reader, and the descriptions are understandable
> - Your feature / bug fix works across **GCP** and **AWS** (where appropriate)
>   - Are modifications to eq-publisher-v3 required?
>   - Are modifications to the Firestore data source required?

---

### What is the context of this PR?

Fixes validation error incorrectly displayed on collection list answers created before the release of repeating answers

https://jira.ons.gov.uk/browse/EAR-2094

### How to review

- Checkout to a branch created before the release of EAR-1772-repeating-individual-answers - do not pull new changes
- Create a questionnaire with a collection list including a list collector answer
- Create a questionnaire with a collection list including no list collector answers
- Create a questionnaire with no collection lists
- Stop running eq-author-api
- Checkout to EAR-2094-list-answer-error
- Re-run eq-author-api
- Check that questionnaires created do not have any incorrect validation errors
- Check `repeatingLabelAndInput` is `false` and `repeatingLabelAndInputListId` is `""` for list answers
- Check all questionnaires created do not include any errors

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
